tests/test_chain_parser.py

```diff
@@ -0,0 +1,46 @@
+from dateutil.parser import ChainParser
+import dateutil.parser
+import pytest
+
+import datetime as dt
+
+
+def custom_parser(dt_str):
+    return dt.datetime(1989, 4, 24)
+
+
+def test_parser_repr():
```

Comment:
I think we maybe don't want to enshrine the exact repr in tests, particularly when it involves the repr of a subclass.

We could maybe mark this as a "smoke test" with `pytest.mark.smoke`, to indicate it's not guaranteed?

Another option might be if we change it so that the repr of `ChainParser("%Y%m%d", "%Y-%m-%d")` is `ChainParser('%Y%m%d', '%Y-%m-%d')`, which seems more consistent with the idea that, if possible, you should be able to copy-paste a repr into a REPL to re-create an equivalent object.